Tests/Shared/UnitTest.swift

```diff
@@ -122,6 +140,10 @@ open class UnitTest: XCTestCase {
         // The SDK does not use `UserDefaults.standard`, but in case a test needs to,
         // let's delete the data for each test.
         UserDefaults.standard.deleteAll()
+
+        keyValueStorage.deleteAll()
```

Comment:
called twice now is that expected?
whats suppose to be cleared by calling this after `switchToGlobalDataStore`?

Reply:
Yes, this is confusing when glancing at it. 

I made a comment explaining what this code is doing. 

`deleteAll()` is needed twice but it deletes a separate set of data.